clients/unshaded/src/main/java/tachyon/client/file/FileOutStream.java

```diff
@@ -120,17 +144,18 @@ public void close() throws IOException {
       if (mCanceled) {
         // TODO(yupeng): Handle this special case in under storage integrations.
         mUnderStorageOutputStream.close();
-        UnderFileSystem underFsClient =
-            UnderFileSystem.get(mUnderStorageFile, ClientContext.getConf());
-        underFsClient.delete(mUnderStorageFile, false);
+        String fileName = PathUtils.temporaryFileName(mFileId, mNonce, mUfsPath);
+        UnderFileSystem underFsClient = UnderFileSystem.get(fileName, ClientContext.getConf());
+        underFsClient.delete(fileName, false);
```

Comment:
Should the fresh fileInfo be gotten from master if the temp file does not exist?